AppKit/CPWindow/CPWindow.j

```diff
@@ -3844,6 +4069,46 @@ var interpolate = function(fromValue, toValue, progress)
 
 @end
 
+@implementation CPWindow (TrackingArea)
+
+- (void)_addToViewsWithTrackingAreas:(CPView)aView
+{
+    [_viewsWithTrackingAreas addObject:aView];
+    
+    var tas = [aView trackingAreas];
+    
+    for (var i = 0, nb = [tas count]; i < nb; i++)
+        [self _addTrackingArea:[tas objectAtIndex:i]];
```

Comment:
Sorry but this would broke the logic.
Even if a view is already in the array, when adding a tracking area to this view, the window must check options and do whatever is needed (also when removing). 
Remember that a view can have multiple tracking areas.
I've tried to put all the logic on the window side but without any information from the view (say which one is the new tracking area), the only solution would consist in adding a tracking areas array (useless and with a negative performances impact).
The _addToViewsWithTrackingAreas (and _remove) are also used in CPView _setWindow.
So (unless I didn't see some evidence) I don't think those modifications are useful...